deeppavlov/models/kbqa/kb_tree.py

```diff
@@ -0,0 +1,362 @@
+                self._relations_mapping = pickle.load(f)
+        if self._templates_filename is not None:
+            with open(self.load_path.parent / self._templates_filename, 'rb') as t:
+                self.templates = pickle.load(t)
+
+    def save(self) -> None:
+        pass
+
+    def __call__(self, sentences: List[str],
+                 *args, **kwargs) -> List[str]:
+
+        objects_batch = []
+        confidences_batch = []
+        for sentence in sentences:
+            is_kbqa = self.is_kbqa_question(sentence)
+            if is_kbqa:
+                q_tokens = nltk.word_tokenize(sentence)
+                entity_from_template, relation_from_template = self.entities_and_rels_from_templates(q_tokens)
+                if entity_from_template and self.use_templates:
+                    if self._debug:
+                        relation_title = self._relations_mapping[relation_from_template]["name"]
+                        log.debug("using templates, entity {}, relation {}".format(entity_from_template,
+                                                                                   relation_title))
+                    entity_triplets, entity_linking_confidences = self.linker(entity_from_template, q_tokens)
+                    rel_prob = 1.0
+                    obj, confidence = self.match_triplet(entity_triplets,
+                                             entity_linking_confidences,
+                                             relation_from_template,
+                                             rel_prob)
+                    objects_batch.append(obj)
+                    confidences_batch.append(confidence)
+                else:
+                    detected_entity, detected_rel = self.entities_and_rels_from_tree(q_tokens)
+                    if detected_entity:
+                        if self._debug:
+                            log.debug("using syntactic tree, entity {}, relation {}".format(detected_entity,
+                                                                                            detected_rel))
+                        entity_triplets, entity_linking_confidences = self.linker(detected_entity, q_tokens)
+                        obj, confidence = self.match_rel(entity_triplets, entity_linking_confidences,
+                                                         detected_rel, sentence)
+                        objects_batch.append(obj)
+                        confidences_batch.append(confidence)
+                    else:
+                        objects_batch.append('')
+                        confidences_batch.append(0.0)
+            else:
+                objects_batch.append('')
+                confidences_batch.append(0.0)
+
+        parsed_objects_batch, confidences_batch = self.parse_wikidata_object(objects_batch, confidences_batch)
+
+        if self.return_confidences:
+            return parsed_objects_batch, confidences_batch
+        else:
+            return parsed_objects_batch
```

Comment:
Maybe we should always return confidences? Different outputs for different cases are weird, and you can just add and ignore confidences in the config.